### DISCUSSION TITLE
Error when installing on Mac

### DISCUSSION THREAD

**Discussion by sandraalsayar** [openai/whisper#2415] [Q&A] [answered]
When I run the command, `pip install -U openai-whisper` or anything else in attempt to install Whisper, I always get this error:

File "/private/var/folders/95/4lj58ngx1nv85r8nrytwwqyw0000gp/T/pip-build-env-xq98ng57/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 318, in run_setup
          exec(code, locals())
          ~~~~^^^^^^^^^^^^^^^^
        File "<string>", line 21, in <module>
        File "<string>", line 11, in read_version
      KeyError: '__version__'
      [end of output]`
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by samarasimhapeyala** [comment, marked as answer]
Instead of installing via pip, clone the latest version of repository, create a virtual env and install requirements via `pip install .` or `pip install -r requirements.txt`
This works!

**Reply by sandraalsayar** [reply]
That didn't work even though i did as you said, cloned the repo, created virtual env inside of it, and typed pip install .
Same error as before i think:
        File "/private/var/folders/95/4lj58ngx1nv85r8nrytwwqyw0000gp/T/pip-build-env-qtj9cnq4/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 319, in run_setup
          exec(code, locals())
          ~~~~^^^^^^^^^^^^^^^^
        File "<string>", line 21, in <module>
        File "<string>", line 11, in read_version
      KeyError: '__version__'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Reply by glangford** [reply]
The issue could be triggered by Python 3.13, have you tried 3.12?

**Reply by sandraalsayar** [reply]
It worked! Thanks!

**Reply by GC0LD** [reply]
im new to using VSC, i have the 3.12 package downloaded on my mac, how to i set that as the interpreter??
been trying to install this and i can only use 3.13 and 3.9, not sure what to do

**Reply by glangford** [reply]
@GC0LD  
StackOverflow - Selecting python interpreter in VSCode
https://stackoverflow.com/questions/76289322/selecting-python-interpreter-in-vscode

**Reply by GC0LD** [reply]
still not figuring it out :( is there any other interface i could use to code it?

**Reply by lawyinking** [reply]
> The issue could be triggered by Python 3.13, have you tried 3.12?

this works for me, successfully installed the open-whisper after downgraded the python version to 3.12
thanks!
